[<libarchive[core] or libarchive[bzip2,core,libxml2,lz4,lzma,lzo,openssl,zstd]>]:x64-windows build failure

**Host Environment**
- OS: [Microsoft Windows [Version 10.0.18363.836]]
- Compiler: Visual Studio 2019 Version: 16.6.0  VS Tools Version: 14.26.28801

**To Reproduce**
Steps to reproduce the behavior:
`.\vcpkg install libarchive`

**Failure logs** 
- Cut and past the appropriate build messages from the console output.
`Computing installation plan...`
`The following packages will be built and installed:`
`    libarchive[bzip2,core,libxml2,lz4,lzma,lzo,openssl,zstd]:x64-windows`
`Starting package 1/1: libarchive:x64-windows`
`Building package libarchive[bzip2,core,libxml2,lz4,lzma,lzo,openssl,zstd]:x64-windows...`
`-- Using cached D:/Git/vcpkg.downloads/libarchive-libarchive-cce09646b566c61c2debff58a70da780b8457883.tar.gz`
`-- Using source at D:/Git/vs2019.prod/buildtrees/libarchive/src/80b8457883-1ab8492808`
`-- Configuring x64-windows`
`-- Building x64-windows-dbg`
`CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):`
`    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v`
`    Working Directory: D:/Git/vs2019.prod/buildtrees/libarchive/x64-windows-dbg`
`    See logs for more information:`
`      D:\Git\vs2019.prod\buildtrees\libarchive\install-x64-windows-dbg-out.log`

`Call Stack (most recent call first):`
`  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)`
`  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)`
`  ports/libarchive/portfile.cmake:67 (vcpkg_install_cmake)`
`  scripts/ports.cmake:90 (include)`

`Error: Building package libarchive:x64-windows failed with: BUILD_FAILED`
`Please ensure you're using the latest portfiles with `.\vcpkg update`, then`
`submit an issue at https://github.com/Microsoft/vcpkg/issues including:`
`  Package: libarchive:x64-windows`
`  Vcpkg version: 2020.02.04-nohash-external`

`Additionally, attach any relevant sections from the log files above.`

- Please attach any additional failure logs mentioned in the console output.
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4660244/install-x64-windows-dbg-out.log)

**Additional context**
1. Note: I only entered vcpkg install libarchive which should have only tried to install the  `core` feature
2. No matter which port features selected it always wants to install all features - reference https://github.com/microsoft/vcpkg/issues/11479 
3. libarchive with all features installs fine with VS2017

```
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(344): error C4061: enumerator 'WT_NONE' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(344): error C4061: enumerator 'WT_INFO' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(344): error C4061: enumerator 'WT_META' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(344): error C4061: enumerator 'WT_REQ' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(344): error C4061: enumerator 'WT_RVIS' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(344): error C4061: enumerator 'WT_CONV' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(344): error C4061: enumerator 'WT_CONT' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(344): error C4061: enumerator 'LAST_WT' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(368): error C4061: enumerator 'WT_NONE' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(368): error C4061: enumerator 'WT_INFO' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(368): error C4061: enumerator 'WT_META' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(368): error C4061: enumerator 'WT_REQ' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(368): error C4061: enumerator 'WT_RVIS' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(368): error C4061: enumerator 'WT_CONV' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(368): error C4061: enumerator 'WT_CONT' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(368): error C4061: enumerator 'LAST_WT' in switch of enum 'warc_type_t' is not explicitly handled by a case label
D:\Git\vs2019.prod\buildtrees\libarchive\src\80b8457883-1ab8492808\libarchive\archive_read_support_format_warc.c(75): note: see declaration of 'warc_type_t'
```

--- a/ports/libarchive/CONTROL
+++ b/ports/libarchive/CONTROL
@@ -1,5 +1,5 @@
 Source: libarchive
-Version: 3.4.1-2
+Version: 3.4.1-3
 Homepage: https://github.com/libarchive/libarchive
 Description: Library for reading and writing streaming archives
 Build-Depends: zlib

--- a/ports/libarchive/disable-c4061.patch
+++ b/ports/libarchive/disable-c4061.patch
@@ -1,0 +1,14 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 911ae5b..0e12b56 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -151,9 +151,6 @@ IF (MSVC)
+   #################################################################
+   # Set compile flags for debug build.
+   # This is added into CMAKE_C_FLAGS when CMAKE_BUILD_TYPE is "Debug"
+-  # Enable level 4 C4061: The enumerate has no associated handler in a switch
+-  #                       statement.
+-  SET(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /we4061")
+   # Enable level 4 C4254: A larger bit field was assigned to a smaller bit
+   #                       field.
+   SET(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /we4254")

--- a/ports/libarchive/portfile.cmake
+++ b/ports/libarchive/portfile.cmake
@@ -12,6 +12,7 @@ vcpkg_from_github(
         fix-lz4.patch
         fix-zstd.patch
         fix-cpu-set.patch
+        disable-c4061.patch
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

